test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlDocumentTest/AcceptsEndTagWithNoMatchingStartTag.stree.txt

```diff
@@ -1,13 +1,14 @@
-Markup block - Gen<None> - 14 - (0:0,0)
-    Markup span - Gen<Markup> - [Foo ] - SpanEditHandler;Accepts:Any - (0:0,0) - Tokens:2
-        SyntaxKind.Text;[Foo];
-        SyntaxKind.Whitespace;[ ];
-    Tag block - Gen<None> - 6 - (4:0,4)
-        Markup span - Gen<Markup> - [</div>] - SpanEditHandler;Accepts:Any - (4:0,4) - Tokens:4
-            SyntaxKind.OpenAngle;[<];
-            SyntaxKind.ForwardSlash;[/];
-            SyntaxKind.Text;[div];
-            SyntaxKind.CloseAngle;[>];
-    Markup span - Gen<Markup> - [ Bar] - SpanEditHandler;Accepts:Any - (10:0,10) - Tokens:2
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Text;[Bar];
+SyntaxKind.HtmlDocument - [0..14) - FullWidth: 14 - Slots: 1 - [Foo </div> Bar]
+    SyntaxKind.HtmlMarkupBlock - [0..14) - FullWidth: 14 - Slots: 1
+        SyntaxKind.HtmlTextLiteral - [0..4) - FullWidth: 4 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+            SyntaxKind.Text;[Foo];
+            SyntaxKind.Whitespace;[ ];
+        SyntaxKind.HtmlTagBlock - [4..10) - FullWidth: 6 - Slots: 1
+            SyntaxKind.HtmlTextLiteral - [4..10) - FullWidth: 6 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                SyntaxKind.OpenAngle;[<];
+                SyntaxKind.ForwardSlash;[/];
+                SyntaxKind.Text;[div];
+                SyntaxKind.CloseAngle;[>];
+        SyntaxKind.HtmlTextLiteral - [10..14) - FullWidth: 4 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
```

Comment:
So an interesteing discussion is whether we want to go with the nomenclature `Html...` or `Markup...`. I like Markup better because it's more general. In the Blazor world we anticipate supporting non-HTML markup int he future. Putting HTML everywhere is a little less good to me. Thoughts?

Reply:
Interesting. Just to clear, do you mean you prefer `Markup..` for everything that is HTML or specifically for a few nodes like `TextLiteral`?

Reply:
Should be consistent. Markup works for me.

Reply:
We'd name it `Blazor` or something equivalent for the type of Markup that we're adding (if we had to). Html just seems too specific because it makes it seem like we understand things like JavaScript, CSS etc. I could get on board for `HtmlStartTag` though (when the time comes) because we know it's Html.